### PR TITLE
[WIP]KAFKA-15444: Native docker image

### PR DESCRIPTION
**Pull request by kagarwal06**
This PR includes the changes for `native-binary` docker image for Apache Kafka corresponding to the KIP: https://cwiki.apache.org/confluence/display/KAFKA/KIP-974%3A+Docker+Image+for+GraalVM+based+Native+Kafka+Broker

**NOTE 1** This PR is dependent on https://github.com/apache/kafka/pull/14552, hence should be merged post merging https://github.com/apache/kafka/pull/14552
**NOTE 2:** The changes are made on top of the https://github.com/apache/kafka/pull/14552 as we had common code. Hence this contains changes for the JVM Docker Image as well
**NOTE 3:** To view Native-Image specific changes, refer: https://github.com/VedarthConfluent/kafka/pull/1

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/Kafka.scala [hunk 1]
```diff
@@ -86,6 +86,10 @@ object Kafka extends Logging {
   }
 
   def main(args: Array[String]): Unit = {
+    process(args)
+  }
+
+  def process(args: Array[String]): Unit = {
```
Comment (ijuma) [inline core/src/main/scala/kafka/Kafka.scala]: Why is `process` needed? You can invoke `main` from `KafkaNativeWrapper`.

File: docker/test/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+confluent_kafka
```
Comment (ijuma) [inline docker/test/requirements.txt]: This doesn't seem right.

Comment (VedarthConfluent) [reply docker/test/requirements.txt]: It seems like it's a typo. Updating it to `confluent-kafka`

Comment (ijuma) [reply docker/test/requirements.txt]: But why would we depend on a confluent kafka artifact.

Comment (VedarthConfluent) [reply docker/test/requirements.txt]: @ijuma Confluent Kafka for Python - https://github.com/confluentinc/confluent-kafka-python
`confluent-kafka` is a python client compatible with kafka
For sanity tests of docker images we wanted a python client to interact with kafka residing in docker image, hence we went ahead with this publicly available and widely used python client.
We are open to consider alternate python clients if there are any concerns in using this library

Comment (ijuma) [reply docker/test/requirements.txt]: Can we not use the CLI tools that are included in the Apache Kafka project?

File: core/src/main/scala/kafka/Kafka.scala (hunk 1 above)
Comment (kagarwal06) [reply core/src/main/scala/kafka/Kafka.scala]: Done

File: docker/test/requirements.txt (hunk 2 above)
Comment (VedarthConfluent) [reply docker/test/requirements.txt]: This suggestion has been incorporated. Now we are using CLI scripts included in kafka.

Comment (ijuma) [reply docker/test/requirements.txt]: Can we remove `confluent_kafka` from here then?

Comment (VedarthConfluent) [reply docker/test/requirements.txt]: It has been removed. These changes are part of KIP 975 PR raised here:- https://github.com/apache/kafka/pull/14552. (commit 6f64ce54254b985a50b3c5d8d63711cc6b71ab18)
They'll reflect here once this PR is updated

Comment (ijuma) [reply docker/test/requirements.txt]: Makes sense, thanks.

**Pull request closed without merging**